Feature/pause menu

RayGar opened the pull request:
Requesting a PR for my feature/pause-menu.

RayGar committed:
Pause menu now displays

Added our first version of the pause menu. Fixed the bug were the pause menu was not displaying. The buttons should be functional, however I do not know how to disable the camera and turn on a cursor so we cannot know if the buttons work until we do that.

RayGar committed:
Updated the GUI for the pause menu

Replaced the old assets so that the pause menu has the same artistic stlyle as the other screens. Tested the buttons' functionality. However, I still have not made the camera stay still at the pause menu.

RayGar committed:
Modified some of the folders in the scene

Forgot to stage a change in the base scene.

RayGar committed:
Fixed a bug

Forgot to restart the timer for the restart function in the pause menu.

RayGar committed:
Replaced the missing logo in the loading screen

RayGar committed:
Revert "Replaced the missing logo in the loading screen"

This reverts commit aa59e437946d04eabe69cf20d4d10a4a9261ea91.

RayGar committed:
Fixed the bugs in the pause menu

zliving reviewed (changes requested):
The screens need to resize with larger displays. Also if you could add both P and ESC as pause buttons.

RayGar committed:
Added functionality to the Escape key

The Escape key now also pauses and unpauses the game.

RayGar committed:
Set the UI Scale Mode to "Scale with screen size"

RayGar committed:
Updated the scaling on the Pause Menu.

Also changed the name of the "Tim-GUI" to "GUI" since that's where I'm gonna have to make all the UI elements a child of.
